agent/consul/discoverychain/gateway.go

```diff
@@ -177,6 +195,27 @@ func (l *GatewayChainSynthesizer) consolidateHTTPRoutes() []structs.HTTPRouteCon
 	return routes
 }
 
+func targetForResolverNode(nodeName string, chains []*structs.CompiledDiscoveryChain) string {
+	resolverPrefix := structs.DiscoveryGraphNodeTypeResolver + ":"
+	splitterPrefix := structs.DiscoveryGraphNodeTypeSplitter + ":"
```

Comment:
Not blocking: Would these make more sense as constants defined at the top of the file?

Reply:
I'm ok with this for now -- most of this code is a temporary holdover as we work towards native xDS for API Gateways anyway, so it'll be dropped soon.